workers/unity/Packages/com.improbable.gdk.core/Tests/Editmode/GameObjectRepresentation/Readers/EventTests.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using Generated.Improbable.Gdk.Tests.BlittableTypes;
+using Generated.Improbable.Gdk.Tests.ComponentsWithNoFields;
+using NUnit.Framework;
+using Unity.Entities;
+using ComponentWithEvents = Generated.Improbable.Gdk.Tests.ComponentsWithNoFields.ComponentWithNoFieldsWithEvents;
+
+namespace Improbable.Gdk.Core.EditmodeTests.MonoBehaviours.Readers
+{
+    [TestFixture]
+    internal class EventTests
+    {
+        protected ComponentWithEvents.Reader ReaderPublic;
+        protected ComponentWithEvents.ReaderWriterImpl ReaderWriterInternal;
+        protected EntityManager EntityManager;
+        protected Entity Entity;
+        private World world;
+
+        [SetUp]
+        public void SetUp()
+        {
+            world = new World("test-world");
+            EntityManager = world.GetOrCreateManager<EntityManager>();
+            Entity = EntityManager.CreateEntity(typeof(SpatialOSBlittableComponent));
```

Comment:
use `ComponentType.Create<>` instead of `typeof`

Reply:
In create entity I think using types is fine. They do almost the same thing and the end result is the same.

Reply:
I agree they do the same thing, But I think not using the implicit conversion is clearer, and allowed for readonly types. Just a good habit to get into?

Reply:
I think the implicit cast exists mostly for this case. Can you have readonly components in an archetype, as that is essentially static data (except for writing directly using the entity manager, which I guess you'd have to do as you don't have default values on an archetype)? Given that most ComponentType modifiers make no sense on an entity I'd be surprised if archetypes respected them, although a quick glance at the code doesn't reveal much.

If you can have readonly types and they are being used then I agree. Otherwise I don't think it matters which one you use, other than how easy it is to read.